[EDU-3627] Bot Protection reference (#538)
aziontech/docs: src/content/docs/pt-br/pages/menu-principal/referencia/marketplace/bot-protection-referencia.mdx

---
title: Azion Bot Protection
description: Proteja suas aplicações contra ataques automatizados, como assinaturas de bots maliciosos ou bots scriptados.
meta_tags: bot, proteção, cybersegurança, edge computing
namespace: docs_bot_protection_reference
permalink: /documentacao/produtos/marketplace/bot-protection/
---

import Button from '~/components/Button.astro';

> O **Bot Protection** da Azion usa algoritmos avançados e *Reputation Intelligence* para analisar o comportamento dos dados recebidos. Isso permite a detecção de tráfego suspeito e bots maliciosos, facilitando a implementação de medidas preventivas contra atividades maliciosas, como preenchimento de credenciais (credential stuffing), varredura de vulnerabilidades (vulnerability scanning,) e raspagem de sites (site scraping).

Author: ```suggestion
O **Bot Protection** da Azion usa algoritmos avançados, machine learning (aprendizado de máquine) e *Reputation Intelligence* para analisar o comportamento dos dados recebidos. Isso permite a detecção de tráfego suspeito e bots maliciosos, facilitando a implementação de medidas preventivas contra atividades maliciosas, como preenchimento de credenciais (credential stuffing), varredura de vulnerabilidades (vulnerability scanning,) e raspagem de sites (site scraping).
```

> O **Bot Protection** da Azion usa algoritmos avançados, machine learning (aprendizado de máquina) e *Reputation Intelligence* para analisar o comportamento dos dados recebidos. Isso permite a detecção de tráfego suspeito e bots maliciosos, facilitando a implementação de medidas preventivas contra atividades maliciosas, como preenchimento de credenciais (credential stuffing), varredura de vulnerabilidades (vulnerability scanning,) e raspagem de sites (site scraping).

Author: ```suggestion
O **Bot Protection** da Azion usa algoritmos inteligentes avançados e *Reputation Intelligence* para analisar o comportamento dos dados recebidos. Isso permite a detecção de tráfego suspeito e bots maliciosos, facilitando a implementação de medidas preventivas contra atividades maliciosas, como preenchimento de credenciais (credential stuffing), varredura de vulnerabilidades (vulnerability scanning,) e raspagem de sites (site scraping).
```

Esses dados podem ser usados para criar regras e comportamentos mais robustos nos argumentos de Bot Protection, a fim de detectar e mitigar ameaças de forma mais eficaz.

### Rate Limiting

O Rate Limiting (limitação de taxa) estabelece limites para o número de requisições que um usuário ou sistema pode fazer dentro de um período de tempo especificado, mitigando efetivamente o impacto de ataques de força bruta ou atividades excessivas de bots. Ao trabalhar em conjunto com limitação de taxa, as medidas do Bot Protection ganham uma camada adicional de defesa contra ameaças automatizadas.

### Fingerprint

Um conjunto de informações (IP, User-Agent) cria um hash para dispositivos que acessam suas edge applications As informações são coletadas rastreando a sessão do dispositivo e fornecem um detalhamento mais preciso do dispositivo usado na requisição, aumentando a precisão dos logs do Bot Protection.

### Captcha 

> Ao usar a ação `redirect`, a URL/localização definida pode conter uma integração de Captcha para adicionar uma camada de segurança adicional. Isso ajuda a aumentar a segurança e a detecção de tráfego malicioso, desafiando todas as requisições que violaram anteriormente qualquer limite para garantir que sejam legítimas.

Author: ```suggestion
Ao usar a ação `redirect`, a URL/localização definida pode conter uma integração de Captcha para adicionar uma camada de segurança adicional. Isso ajuda a aumentar a segurança e a detecção de tráfego malicioso, apresentando um desafio para todas as requisições que violaram anteriormente qualquer limite para garantir que sejam legítimas.
```

### Custom rules

> A Azion fornece configurações fáceis de usar, que devem ser suficientes para a maioria dos casos. Se você precisar de uma configuração mais detalhada, poderá editar o arquivo JSON da integração para personalizar e adicionar novas regras com base nas necessidades do seu negócio. Também é possível adicionar mais critérios e comportamentos a serem executados pelo [Rules Engine](/en/documentation/products/edge-firewall/rules-engine/), construindo respostas mais abrangentes para possíveis ataques.

Author: ```suggestion
A Azion fornece configurações fáceis de usar, que devem ser suficientes para a maioria dos casos. Se você precisar de uma configuração mais detalhada, poderá editar o arquivo JSON da integração para personalizar e adicionar novas regras com base nas necessidades do seu negócio. Também é possível adicionar mais critérios e comportamentos a serem executados pelo [Rules Engine](/pt-br/documentacao/produtos/edge-firewall/rules-engine/), construindo respostas mais abrangentes para possíveis ataques.
```

---
title: Azion Bot Protection
description: Proteja suas aplicações contra ataques automatizados, como assinaturas de bots maliciosos ou bots scriptados.
meta_tags: bot, proteção, cybersegurança, edge computing
namespace: docs_bot_protection_reference
permalink: /documentacao/produtos/marketplace/bot-protection/
---

import Button from '~/components/Button.astro';

O **Bot Protection** da Azion usa algoritmos inteligentes avançados e *Reputation Intelligence* para analisar o comportamento dos dados recebidos. Isso permite a detecção de tráfego suspeito e bots maliciosos, facilitando a implementação de medidas preventivas contra atividades maliciosas, como preenchimento de credenciais (credential stuffing), varredura de vulnerabilidades (vulnerability scanning,) e raspagem de sites (site scraping).

Esta integração atribui uma pontuação a cada requisição com base em regras, comportamentos e Reputation Intelligence. Se a pontuação exceder um limite predeterminado (threshold), a integração executa a ação predeterminada para negar (deny), descartar (drop) ou redirecionar (redirect) a requisição.

> Al usar **Bot Protection**, você pode:

Reviewer: ```suggestion
Ao usar **Bot Protection**, você pode:
```

- **Melhorar a experiência do usuário**
    - Reduzindo o impacto dos bots em toda a infraestrutura.
    - Fornecendo proteção contra bots por reputação de IP.
    - Definindo gerenciamento de regras personalizadas para agir em bots individuais com base na atividade de extração de conteúdo anterior das requisições.

- **Aumentar a visibilidade**
    - Medindo a quantidade e as características do tráfego de bots tentando acessar seu site, APIs e aplicações.

>     - Exportando seus logs, integrando seu SIEM ou usando as ferramentas de observabilidade fornecidas pela Azion para monitorar a atividade maliciosa.

Reviewer 2: ```suggestion
    - Exportando seus logs, integrando seu Gerenciamento e Correlação de Eventos de Segurança (SIEM) ou usando as ferramentas de observabilidade fornecidas pela Azion para monitorar a atividade maliciosa.
```

Tbm não sei se precisa!

>     - Combinando outras integrações para melhorar a eficiência de Bot Protection, através do uso de fingerprint, captchas, injeção de JavaScript, ou SDKs, para criar regras robustas.

Reviewer: ```suggestion
    - Combinando outras integrações para melhorar a eficiência de Bot Protection, através do uso de fingerprint, captchas, injeção de JavaScript ou SDKs, para criar regras robustas.
```

- **Reduzir o risco financeiro**
    -  Protegendo seu site e aplicações contra abuso de credenciais, verificação de saldo de cartão e outras formas de fraude online.

---

## Implementação

| Escopo| Descrição |
|---|---|
| [Como instalar o Azion Bot Protection](/pt-br/documentacao/produtos/guias/bot-protection/) | Guia passo a passo para instalar e configurar a integração Bot Protection. |
| [Conheça as Azion Integrations](/pt-br/documentacao/produtos/marketplace/integracoes/) | Referência para aprofundar os conceitos básicos das integrações da Azion, como elas funcionam e como você pode começar a usá-las. |

---

## Como funciona o Bot Protection da Azion

O **Bot Protection** da Azion é uma integração *serverless* disponível no **Marketplace** da Azion.

Esta integração é baseada em uma edge function que é executada no [Edge Firewall](/pt-br/documentacao/produtos/edge-firewall/). Esse tipo de função aproveita os recursos e módulos do **Edge Firewall** para proteger sua rede, aplicações e recursos contra ameaças de segurança, incluindo assinaturas de bots maliciosos ou bots scriptados.

/// GRÁFICO ///

> ////

Reviewer: rever

1. Uma requisição chega a um domínio usando Bot Protection.
2. O Edge Firewall recebe a requisição.
3. A função é executada e inicia todos os processos de análise, incluindo:

>     - Coleta dos dados da srequisição, incluindo dispositivo, navegador e dados de rede, fingerprint, entre outros.

Author: ```suggestion
    - Coleta dos dados da requisição, incluindo dispositivo, navegador e dados de rede, fingerprint, entre outros.
```

Para aprimorar ainda mais suas capacidades de proteção, ela permite a incorporação de camadas de segurança adicionais por meio de outras integrações e recursos, como [SDKs](#sdks), [injeção de JavaScript](#injecao-de-javascript) e [Fingerprint](#fingerprint) para coletar dados mais granulares.

### Redirecionamento

Uma das ações que o Bot Protection é capaz de executar é o `redirect`, ou redirecionamento. Isto permite que a requisição seja redirecionada para uma nova URL/localização, especificada nos argumentos JSON, quando o limite de segurança é atingido.

### HTML personalizado

A integração permite a entrega de conteúdo HTML personalizado ao usuário em caso de violação do limite, através da ação `custom_html`. Você pode criar uma mensagem personalizada para exibir aos usuários em caso de violação do limite.

### Delay response

A integração permite a manutenção de uma conexão aberta e o atraso nas respostas nos casos em que os bots precisam funcionar autonomamente por períodos prolongados. Isso elimina a necessidade de respostas contínuas, otimizando as operações e garantindo desempenho e confiabilidade consistentes em tarefas críticas.

### Modos

> O **Bot Protection** permite que você defina o ambiente no qual a função deve ser executada, sendo **API** ou **Web** os modos possíveis. O modo padrão é `web`. Se for fornecido um valor diferente da string "api" (sensível a maiúsculas e minúsculas), o modo "web" será usado como configuração padrão.

Reviewer 2: ```suggestion
O **Bot Protection** permite que você defina o ambiente no qual a função deve ser executada, sendo **API** ou **Web** os modos possíveis. O modo padrão é `web`. Se for fornecido um valor diferente da string `api` (sensível a maiúsculas e minúsculas), o modo `web` será usado como configuração padrão.
```

Ao habilitar o modo `api`, nenhum "Set-Cookie" será realizado, e quaisquer regras relacionadas ao uso de cookies no Bot Protection serão ignoradas.

### Exportação de logs

Ao exportar um arquivo JSON com logs para um endpoint HTTPS, você pode manter um registro detalhado das interações, capacitando-lhe com as informações necessárias para tomar decisões informadas e aprimorar continuamente a postura de segurança de seus sistemas.

Quando um log é gerado, você receberá um objeto JSON semelhante a este:

```json
{
  "geoip_region": "SP",
  "action": "deny",
  "asn": "396982",
  "request_id": "9581e2b51b5a082b12fed308f4eae564",
  "host": "ojgayey.map.azionedge.com",
  "host": "xxxxxxx.map.azionedge.com",
  "classified": "bad bot",
  "geoip_country": "BR",
  "http_user_agent": "curl/7.87.0",
  "bot category": [
    "Bad Bot Signatures"
  ],
  "request_uri": "/a",
  "remote_addr": "34.95.175.91",
  "score": 8
}
```

> Esta funcionalidade fornece monitoramento abrangente, análise pós-evento e rastreamento em tempo real de ameaças potenciais ou atividades suspeitas. Você pode usar seu endpoint preferido, incluindo endpoints do **Data Streaming** da Azion e SIEMs, para receber as informações.

Reviewer: corrigir em EN também, nem todo SIEM é real time

> Com os endpoints do Data Streaming, você pode alimentar seus sistemas de processamento de fluxo, SIEM e plataformas de big data com os logs de eventos de suas aplicações na Azion em tempo real. Este produto permite que você colete seus logs e os conecte ao endpoint de suas plataformas de análise de dados. Você também pode decidir quais dados deseja usar em sua análise, escolhendo entre as variáveis disponíveis.

Reviewer: ```suggestion
Com os conectores do Data Streaming, você pode alimentar seus sistemas de processamento de fluxo, SIEM e plataformas de big data com os logs de eventos de suas aplicações na Azion em tempo real. Este produto permite que você colete seus logs e os conecte ao endpoint de suas plataformas de análise de dados. Você também pode decidir quais dados deseja usar em sua análise, escolhendo entre as variáveis disponíveis.
```

<Button href="/pt-br/documentacao/produtos/data-streaming/" text="saiba mais sobre o data streaming" variant="secondary"> 

</Button>

---

## Recursos adicionais

### SDKs

> Azion **Bot Protection** can work altogether with Software Development Kits (SDKs), for both Android and iOS systems, allowing you to customize and tailor security protocols to meet the specific needs of your mobile applications. With SDKs and Bot Protection, you can implement fine-grained controls, address application-specific vulnerabilities, and adapt to evolving threats more effectively.
> 
> You can use SDKs to track mobile devices and identify behaviors (touching the screen, for example) and device data (model, manufacturer, operational system, etc.) to use as insights for Bot Protection detect and mitigate malicious threats. 

Reviewer 2: ```suggestion
Azion **Bot Protection** pode trabalhar em conjunto com Software Development Kits (SDKs), tanto para sistemas Android quanto iOS, permitindo que você personalize e ajuste os protocolos de segurança para atender às necessidades específicas de suas aplicações móveis. Com SDKs e Bot Protection, você pode implementar controles refinados, abordar vulnerabilidades específicas da aplicação e adaptar-se a ameaças em evolução de forma mais eficaz.

Você pode usar SDKs para rastrear dispositivos móveis e identificar comportamentos (como toque na tela) e dados do dispositivo (modelo, fabricante, sistema operacional, etc.) para usar como insights para que o Bot Protection detecte e mitigue ameaças maliciosas.
```
